src/get_ensemble_output.R

```diff
@@ -1,30 +1,62 @@
-library(dplyr)
-library(ggplot2)
 library(hubUtils)
 library(hubVis)
 library(hubEnsembles)
+library(hubData)
 
-model_names = c("epiENGAGE-GBQR", "epiENGAGE-Copycat", "epiENGAGE-INFLAenza", "epiforecasts-dyngam")
-reference_date = "2025-03-15"
 
-for(i in 1:length(model_names)){
-  assign("tmp", read.csv(paste("https://raw.githubusercontent.com/reichlab/flu-metrocast/main/model-output/", model_names[i], "/", reference_date, "-", model_names[i], ".csv", sep="")))
-  assign(paste("output",i, sep=""), get("tmp") %>% mutate(model_id = model_names[i]))
-}
+library(gh)
+library(purrr)
+library(readr)
+library(dplyr)
+
+# GitHub API request to list files in the "model-output" folder
+repo <- "reichlab/flu-metrocast"
```

Comment:
I feel strongly that we should be using the hubData functionality to read the files here. For now, this would involve ["connecting" to a local version of the hub](https://hubverse-org.github.io/hubData/articles/connect_hub.html#connecting-to-a-local-hub) and then using that connection to run a `collect_hub()` call, as documented in that vignette. 

Relatedly, we can/should set up an s3 bucket for these data if we haven't already. And then you could use hubData to read data from that cloud service directly, without having to assume any local version of the repo is present.